source/fab/cli.py

```diff
@@ -0,0 +1,83 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict, Optional
+
+import fab
+from fab.artefacts import CollectionGetter
+from fab.build_config import BuildConfig
+from fab.constants import PRAGMAD_C
+from fab.steps.analyse import Analyse
+from fab.steps.c_pragma_injector import CPragmaInjector
+from fab.steps.compile_c import CompileC
+from fab.steps.compile_fortran import CompileFortran, get_fortran_compiler, get_fortran_preprocessor
+from fab.steps.find_source_files import FindSourceFiles
+from fab.steps.grab.folder import GrabFolder
+from fab.steps.link import LinkExe
+from fab.steps.preprocess import c_preprocessor, fortran_preprocessor
+from fab.steps.root_inc_files import RootIncFiles
+
+
+def _generic_build_config(folder: Path, kwargs: Optional[Dict] = None) -> BuildConfig:
+    folder = folder.resolve()
+    kwargs = kwargs or {}
+
+    # Within the fab workspace, we'll create a project workspace.
+    # Ideally we'd just use folder.name, but to avoid clashes, we'll use the full absolute path.
+    label = '/'.join(folder.parts[1:])
+
+    fpp, fpp_flags = get_fortran_preprocessor()
+    fc, fc_flags = get_fortran_compiler()
+
+    if fc == 'gfortran':
+        link_step = LinkExe(linker='gcc', flags=['-lgfortran'])
+    else:
+        raise NotImplementedError(f"Fab's zero config not yet configured for compiler: '{fc}'")
```

Comment:
In my "mesh tools" script I made use of a dictionary to easily map compiler to argument list. Would that approach make sense here? e.g.

```python
fortran_linker = {
    'gfortran': ('gcc, ['-lgfortran']),
    'ifort': (..., [...])
}
...
try:
    link_step = LineExe(linker=fortran_link[fc][0], flags=fortran_link[fc][1])
except KeyError:
    raise NotImplementedError(f"Fab's zero configuration mode does not yet work with compiler '{fc}'"
```

Reply:
Yes, that makes perfect sense. It's the natural progression as we add more compilers but we might as well do that right away.